src/main/scala/com/vngrs/etl/Combine.scala

```diff
@@ -0,0 +1,49 @@
+package com.vngrs.etl
+
+import org.apache.spark.rdd.RDD
+
+import scala.reflect.ClassTag
+
+/**
+  * [[com.vngrs.etl.Combine]] is an interface for combining two [[com.vngrs.etl.Pipeline]]s into one.
+  *
+  * @tparam A Type of Input [[com.vngrs.etl.Pipeline]]
+  * @tparam B Type of Output [[com.vngrs.etl.Pipeline]]
+  */
+trait Combine[A, B] {
```

Comment:
Well, this is the only interface that can handle two `Pipeline` and accepted and used by `Pipeline` itself. For now its only usage is `Zip` and `Union` and I agree for just them it is not required to have a interface like this. However, we should keep it in the beginning and see how things will go. If it stays like this, we can make it deprecated.